quipucords/api/scan/view.py

```diff
@@ -133,6 +138,95 @@ def expand_scan(json_scan):
     return json_scan
 
 
+def _destroy_scan(scan: Scan, sender=None) -> None:
+    """Delete a scan, its jobs, and results tasks.
+
+    :param scan: the Scan to be deleted
+    :param sender: optional sender for cancel_scan signal
+    """
+    logger.info("Deleting scan jobs associated with scan %s", scan.id)
+    if scan.jobs is not None:
+        jobs_to_cancel = scan.jobs.exclude(
+            Q(status=ScanTask.FAILED)
+            | Q(status=ScanTask.CANCELED)
+            | Q(status=ScanTask.COMPLETED)
+        )
+        if jobs_to_cancel:
```

Comment:
uh, never did this before!

is this equivalent to `jobs_to_cancel.exists()`? or just makes the query and is truth if it has results?

Reply:
I'll be honest here: I didn't write this code and I didn't pay much attention to it! 😅  I just cut and paste from the method `ScanViewSet.destroy` to this new function `_destroy_scan` so that my new code could perform the same operations and stay DRY. ☀️

That being said, yes, it seems like this does the latter: execute the query and be truthy if it produces any results. It's not recommended, though, if you don't also want to iterate through the results. From the [Django docs](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#when-querysets-are-evaluated):

> Note: If you only want to determine if at least one result exists (and don’t need the actual objects), it’s more efficient to use [exists()](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#django.db.models.query.QuerySet.exists).

Fortunately, in this case, we _do_ want to iterate the results in most cases, and this is fine in terms of efficiency.